lib/archethic/contracts/interpreter/library.ex

```diff
@@ -217,4 +195,33 @@ defmodule Archethic.Contracts.Interpreter.Library do
       bin
     end
   end
+
+  @doc """
+  Get the genesis address of the chain
+  """
+  @spec get_genesis_address(binary()) ::
+          binary()
+  def get_genesis_address(address) do
+    addr_bin = decode_binary(address)
+
+    case TransactionChain.fetch_genesis_address_remotely(addr_bin) do
+      {:ok, genesis_address} -> Base.encode16(genesis_address)
+      {:error, _} -> address
+    end
+  end
+
+  @doc """
+  Get the First transaction address of the transaction chain for the given address
+  """
+  @spec get_first_transaction_address(address :: binary()) ::
+          binary()
+  def get_first_transaction_address(address) do
+    addr_bin = decode_binary(address)
+
+    case TransactionChain.fetch_first_transaction_address_remotely(addr_bin) do
+      {:ok, ^address} -> address
```

Comment:
be careful, here `address` is not always a hexadecimal. Example: `contract.address` will return a binary